src/network/thrift_server.cpp

```diff
@@ -86,30 +86,34 @@ class InfinityServiceHandler : virtual public infinity_thrift_rpc::InfinityServi
     void Connect(infinity_thrift_rpc::CommonResponse &response) override {
         auto infinity = Infinity::RemoteConnect();
         if (infinity == nullptr) {
-            response.success = false;
-            response.error_msg = "Connect failed";
+            response.__set_success(false);
+            response.__set_error_code((i64)(ErrorCode::kConnectFailed));
+            response.__set_error_msg("Connect failed");
             LOG_ERROR(fmt::format("THRIFT ERROR: Connect failed"));
         } else {
             std::lock_guard<std::mutex> lock(infinity_session_map_mutex_);
             infinity_session_map_.emplace(infinity->GetSessionId(), infinity);
-            response.session_id = infinity->GetSessionId();
-            response.success = true;
+            response.__set_session_id(infinity->GetSessionId());
+            response.__set_success(true);
+            response.__set_error_code((i64)(ErrorCode::kOk));
         }
     }
 
     void Disconnect(infinity_thrift_rpc::CommonResponse &response, const infinity_thrift_rpc::CommonRequest &request) override {
         auto infinity = GetInfinityBySessionID(request.session_id);
```

Comment:
Create a new function: GetAndRemoveSessionID:
lock the mutex and try to remove the session id. If no session id is found, return error Status.